# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/base/LocationBase.cpp
# Commit: d0acf898fb0d
# Full Hash: d0acf898fb0d00fac535141d777659ef6702b710
# Author: Kris Maglione <maglione.k@gmail.com>
# Date: 2019-09-12 21:52:31
# Description:
#   Bug 1574593: Silently ignore load attempt on/from discarded BrowsingContext. r=nika
#   
#   The (non-normative) window.open spec does not specify what should happen when
#   window.open is called on a window with a null/discarded browsing context, but
#   in general the lookup and creation rules do not make sense when the window has
# ==============================================================================

diff -r d54701f32f28 -r d0acf898fb0d dom/base/LocationBase.cpp
--- a/dom/base/LocationBase.cpp	Thu Sep 12 14:35:46 2019 +0300
+++ b/dom/base/LocationBase.cpp	Thu Sep 12 11:37:01 2019 +0000
@@ -14,6 +14,7 @@
 #include "nsError.h"
 #include "nsContentUtils.h"
 #include "nsGlobalWindow.h"
+#include "mozilla/NullPrincipal.h"
 #include "mozilla/dom/Document.h"
 
 namespace mozilla {